### DISCUSSION TITLE
Is it possible to customize better-branch with extra segments?

### DISCUSSION THREAD

**Discussion by groovyghoul** [Everduin94/better-commits#68] [Q&A]
Was just wondering if it is possible to prompt and build out an extra segment via configuration. For example, better-branch currently makes the following branch name for me, `user/feat/tix-44506-update-level`, but I would like to add a segment for version number (of the branch that I am coming off of). For example, `user/8.87/feat/tix-44506-update-level`.

Any possibility? Even without this, I absolutely love this program. Thanks.

**Comment by Everduin94** [comment]
@groovyghoul - Hey thanks for the kind words, I appreciate it!

Let me think about how to solve that. I wouldn't mind adding something that makes better-branch more configurable. Maybe some sort of template property. But, that might take me a while to get to. I'm currently on paternity leave and my baby is a handful 😅. 

Is adding version number like that something your work does? A common pattern? Something you personally prefer?

**Comment by groovyghoul** [comment]
Yeah, it is a convention that we use at work (and a personal preference). That way if we have 3 difference release versions being worked on (say a release for 8.88, a patch for 8.87 and another patch for 8.86), you can tell where you came off of by looking at the branch name easily.

user/8.88/feat/new-feature --> a new feature being worked on to put back into the release branch
user/8.86/fix/squash-dat-bug --> fixing a bug in patch 8.86

This allows devs to flip around throughout the day, depending on interruptions, priority, etc. and the branch name now has a lot more meaning to us, since you could be working on multiple things "at the same time"  😄 

I have looked at the code-base and it seems like it would be fairly easy for me to jam in what I want for my own usage, but a template property sounds more extendable for other possible cases.

Congratulations on the new baby! It is the start of a great journey. My first just graduated from university (valedictorian of the class of 2023...Dad-brag!! hahaha) and the second is pretty much ready to start. Good luck!

**Reply by Everduin94** [reply]
Yeah I was going to say it would be quite easy to just add as an optional config. I ask about it's origin because if enough people use it (and in the same order) I think it's an easy addition that makes a lot of sense. My previous work uses version number like this to trigger specific builds, but placed in a different order. Hence why I mention the templating. :)

Also congrats on the valedictorian! Love the dad-brag hahah
